pkg/app/app.go

```diff
@@ -536,6 +536,10 @@ func (h HTTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		log.Println("Error: ", err)
 		var e *apperr.AppError
 		if errors.As(err, &e) {
+			if e.Error() == accounts.MissingUsername {
```

Comment:
can this be rather added in the `Authenticate` method handler above? there we can make sure this condition is only added if using the `UsernameOnly` account manager.

Reply:
I moved this redirection to the `Authenticate` method. Also, I think if we put the redirection logic there, we can redirect when receiving a `nil` user from `UsernameOnlyAccountManager`. It means that we don't need to define a new error type. WDYT?

Reply:
LGTM